app/actions/views/channel.js

```diff
@@ -655,13 +655,15 @@ export function loadChannelsForTeam(teamId, skipDispatch = false) {
             for (let i = 0; i <= MAX_RETRIES; i++) {
                 try {
                     console.log('Fetching channels attempt', teamId, (i + 1)); //eslint-disable-line no-console
-                    const [channels, channelMembers] = await Promise.all([ //eslint-disable-line no-await-in-loop
+                    const [channels, channelMembers, myGroups] = await Promise.all([ //eslint-disable-line no-await-in-loop
                         Client4.getMyChannels(teamId, true),
                         Client4.getMyChannelMembers(teamId),
+                        Client4.getGroupsByUserId(currentUserId),
```

Comment:
super quick, before I continue the review... this API is available in all servers? I guess not! if not available this will return an error and the Promise will fail causing issues with `getMyChannels` and `getMyChannelMembers`

Reply:
I put it in the load groups data function after the server check and license check and team check.
I had to await it after the loop because I kept getting the error [no-await-in-loop](https://eslint.org/docs/rules/no-await-in-loop).